pkg/services/cloudmigration/cloudmigrationimpl/xorm_store_test.go

```diff
@@ -282,6 +282,49 @@ func Test_SnapshotResources(t *testing.T) {
 	})
 }
 
+func TestGetSnapshotList(t *testing.T) {
+	t.Parallel()
+
+	_, s := setUpTest(t)
+	// Taken from setUpTest
+	sessionUID := "qwerty"
+	ctx := context.Background()
+
+	t.Run("returns list of snapshots that belong to a session", func(t *testing.T) {
+		snapshots, err := s.GetSnapshotList(ctx, cloudmigration.ListSnapshotsQuery{SessionUID: sessionUID, Page: 1, Limit: 100})
+		require.NoError(t, err)
+
+		ids := make([]string, 0)
+		for _, snapshot := range snapshots {
+			ids = append(ids, snapshot.UID)
+		}
+		slices.Sort(ids)
+
+		// There are 3 snapshots in the db but only 2 of them belong to this specific session.
+		assert.Equal(t, []string{"lkjhg", "poiuy"}, ids)
+	})
+
+	t.Run("only the snapshots that belong to a specific session are returned", func(t *testing.T) {
+		snapshots, err := s.GetSnapshotList(ctx, cloudmigration.ListSnapshotsQuery{SessionUID: "session-uid-that-doesnt-exist", Page: 1, Limit: 100})
+		require.NoError(t, err)
+		assert.Empty(t, snapshots)
+	})
+
+	t.Run("if the session is deleted, snapshots can't be retrieved anymore", func(t *testing.T) {
+		// Delete the session.
+		_, _, err := s.DeleteMigrationSessionByUID(ctx, sessionUID)
```

Comment:
this deletes the snapshots now,  so this test might not be doing what is intended. Maybe there is a way to delete the session record directly in the database